### PR TITLE
connect to existing executor instance

### PR DESCRIPTION
**Pull request by weswalla**
TODO:
- [x] perspect3ve connects to ad4min instead of starting its own executor
- [x] perspect3ve can still spawn its own executor during dev cycle via npm script
- [x] perspect3ve release builds without executor can be created easily via npm script
- [x] perspect3ve knows on which port ad4min has spawned the executor

- [ ] error handling when executor port not found
- [x] agent initialization and unlock handling from perspect3ve

### CODE REVIEW

**Comment by weswalla** [comment]
@lucksus @fayeed what is a good flow for checking if `executor-port` exists in the app data directory?
- in `main.js`, check if file exists, if it does read the port and save to json where it can be accessed by `App.svelte`
- if doesn't exists, inform user to ensure ad4m-executor is running and close app

**Comment by fayeed** [comment]
@weswalla You could have an ipcMain event in electron which will find the file & send the port back to the client, if not show an error dialog or send the error back to the client app. You can then make the call from the client when the app mounts.

**Comment by weswalla** [comment]
@lucksus @fayeed @kaichaosun this is ready for review/feedback. Could definitely include better error handling, and also I started looking into app building, but maybe we want that to be a separate PR? and also include CI automated build process?

File: executorConfig.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (fayeed) [inline executorConfig.json:1]: Not using this file right?

File: main.js [hunk 2]
```diff
@@ -16,108 +16,145 @@ let bootstrapFixtures = {
...
+        createWindow()
+        splash.close()
+      });
     });
-  });
-
+  }
+  else {
+    try {
+      executorPort = fs.readFileSync(path.join(app.getPath("appData"), '../.local/share/ad4m/executor-port'), {encoding:'utf8', flag:'r'})
```
Comment (fayeed) [inline main.js:137]: You could use this package https://github.com/perspect3vism/appdata-path to get the path, would work on all platforms.

File: main.js [hunk 3]
```diff
@@ -16,108 +16,145 @@ let bootstrapFixtures = {
...
-              "proof": {
-                "signature": "88269fae7990dbd2fdbeb11431333120bc9bd49ae7c7619c19990dce4fca2f054fea4c6da0f331fb14a1be7c6218732c64da24969a551174f603f487290e30ad",
-                "key": "#zQ3shkkuZLvqeFgHdgZgFMUx8VGkgVWsLA83w2oekhZxoCW2n",
-                "valid": true
+  let executorPort
+  if (spawnExecutor) {
+    executorPort = 4000
+    ad4m
+    .init({
+      appDataPath: app.getPath("appData"),
```
Comment (fayeed) [inline main.js:35]: Same here could use the below package.

File: main.js [hunk 4]
```diff
@@ -16,108 +16,145 @@ let bootstrapFixtures = {
   worldLinkLinguageBundle:  fs.readFileSync(path.join('./bootstrap/', worldLinkLanguageHash, 'bundle.js')),
   worldLinkLinguageMeta: JSON.parse(fs.readFileSync(path.join('./bootstrap/', worldLinkLanguageHash, 'meta.json'))),
 }
+const {ipcMain} = require('electron');
+const { exit } = require("process");
+
+let spawnExecutor = false
+console.log(process.argv)
+if(process.argv.length === 4 && process.argv[2] === "executor") {
```
Comment (fayeed) [inline main.js:24]: Do we need to pass executor as command-line argument? We could check if the app is in production using `app.isPackaged`.

**Review by lucksus — APPROVED** [review]

**Pull request merged**